pkg/gatewaysecret/handler.go

```diff
@@ -0,0 +1,137 @@
+package gatewaysecret
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	certmanagerv1 "github.com/cert-manager/cert-manager/pkg/apis/certmanager/v1"
+	"github.com/go-logr/logr"
+	apicorev1 "k8s.io/api/core/v1"
+	apimetav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/fields"
+	"k8s.io/apimachinery/pkg/watch"
+	"k8s.io/client-go/kubernetes"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/kyma-project/lifecycle-manager/internal/pkg/flags"
+)
+
+const (
+	GatewaySecretName = "istio-gateway-secret"
+	kcpRootSecretName = "klm-watcher"
+	kcpCACertName     = "klm-watcher-serving"
+	istioNamespace    = flags.DefaultIstioNamespace
+)
+
+type GatewaySecretHandler struct {
+	*secretManager
```

Comment:
Why need this wrapper struct only for `secretManager`, can you just use secretManager do all the thing?